data-runtime/src/main/java/io/micronaut/data/runtime/operations/internal/sql/DefaultSqlPreparedQuery.java

```diff
@@ -157,6 +169,30 @@ public void attachPageable(Pageable pageable, boolean isSingleResult) {
             SqlQueryBuilder queryBuilder = sqlStoredQuery.getQueryBuilder();
             StringBuilder added = new StringBuilder();
             Sort sort = pageable.getSort();
+            if (pageable instanceof CursoredPageable cursored) {
+                // Create a sort for the cursored pagination. The sort must produce a unique
+                // sorting on the rows. Therefore, we make sure id is present in it.
+                List<Order> orders = new ArrayList<>(sort.getOrderBy());
+                List<RuntimePersistentProperty<E>> idProperties;
```

Comment:
Check new `getIdentityProperties` that can replace this check

Reply:
The method only exists in the `4.7.x` branch. This is the `4.8.x` version: https://github.com/micronaut-projects/micronaut-data/blob/8d204bd69fc6b60d795b86d341d8a46ecea0b923/data-model/src/main/java/io/micronaut/data/model/runtime/RuntimePersistentEntity.java#L297-L307.